src/client/editor/render.js

```diff
@@ -1,156 +1,24 @@
+	map = new renderCore.MarbleMap({
+		world: {
+			waterLevel: -9,
+			sunInclination: .25
 		}
 	});
 
-	editorLog("Renderer loaded");
-
-	animate();
-}
-
-let ambientLight = new THREE.AmbientLight( 0x746070 );
-scene.add( ambientLight );
-
-// Default grid & axes
-
-let axesHelper = new THREE.AxesHelper( 3 );
-scene.add( axesHelper );
-
-let gridHelper = new THREE.GridHelper( 20, 20 );
-scene.add( gridHelper );
+let gridHelper = new THREE.GridHelper(20, 20);
+map.scene.add(gridHelper);
 gridHelper.position.y = -.01;
 
-// Sun
-
-let light = new THREE.DirectionalLight( 0xf5d0d0, 1.5 );
-light.castShadow = true;
-scene.add( light );
-
-// Water
-let waterGeometry = new THREE.PlaneBufferGeometry( 10000, 10000 );
-
-let water = new THREE.Water(
-	waterGeometry,
-	{
-		textureWidth: 512,
-		textureHeight: 512,
-		waterNormals: new THREE.TextureLoader().load( "resources/textures/waternormals.jpg", function( texture ) {
-			texture.wrapS = texture.wrapT = THREE.RepeatWrapping;
-		}),
-		alpha: 1.0,
-		sunDirection: light.position.clone().normalize(),
-		sunColor: 0xffffff,
-		waterColor: 0x001e0f,
-		distortionScale:  3.7,
-		fog: scene.fog !== undefined
-	}
-);
-
-water.rotation.x = - Math.PI / 2;
-water.position.y = -9;
-water.material.uniforms.size.value = 8;
-
-scene.add( water );
-
-// Skybox
-
-let sky = new THREE.Sky();
-sky.scale.setScalar( 10000 );
-scene.add( sky );
-
-let uniforms = sky.material.uniforms;
-
-uniforms.turbidity.value = 10;
-uniforms.rayleigh.value = 2;
-uniforms.luminance.value = 1;
-uniforms.mieCoefficient.value = 0.005;
-uniforms.mieDirectionalG.value = 0.8;
-
-let sunParameters = {
-	distance: 4000,
-	inclination: 0.49,
-	azimuth: 0.205
-};
-
-let cubeCamera = new THREE.CubeCamera( 1, 20000, 256 );
-cubeCamera.renderTarget.texture.minFilter = THREE.LinearMipMapLinearFilter;
-
-function updateSun() {
-	let theta = Math.PI * ( sunParameters.inclination - 0.5 );
-	let phi = 2 * Math.PI * ( sunParameters.azimuth - 0.5 );
-
-	light.position.x = sunParameters.distance * Math.cos( phi );
-	light.position.y = sunParameters.distance * Math.sin( phi ) * Math.sin( theta );
-	light.position.z = sunParameters.distance * Math.sin( phi ) * Math.cos( theta );
-
-	sky.material.uniforms.sunPosition.value = light.position.copy( light.position );
-	// light.shadow.camera.position.copy(light.position);
-	// light.shadow.camera.rotation.copy(light.rotation);
-	light.shadow.mapSize.width = 2048;  // default
-	light.shadow.mapSize.height = 2048; // default
-	light.shadow.camera.near = 3500;
-	light.shadow.camera.far = 4200;
-	light.shadow.camera.left = -60;
-	light.shadow.camera.right = 60;
-	light.shadow.camera.top = 50;
-	light.shadow.camera.bottom = -30;
-	water.material.uniforms.sunDirection.value.copy( light.position ).normalize();
-
-	cubeCamera.update( renderer, scene );
-}
-
-//
-
-uniforms = {
-	time: { value: 1.0 }
-};
-let clock = new THREE.Clock();
+map.addToWorld();
 
 // Default model
 let defaultModel;
```

Comment:
Like mentioned in `client/client/render.js`, maybe have the default model as a global fallback model in render core?